subprojects/reporting/src/main/java/org/gradle/api/reporting/internal/SimpleReport.java

```diff
@@ -53,11 +58,12 @@ public String toString() {
     }
 
     public File getDestination() {
-        return destination == null ? null : resolveToFile(destination);
+        Object evaluatedDestination = destination.isPresent() ? destination.get() : null;
+        return evaluatedDestination == null ? null : resolveToFile(evaluatedDestination);
     }
 
-    protected void setDestination(Object destination) {
-        this.destination = destination;
+    public void setDestination(Object destination) {
+        this.destination.set(destination);
```

Comment:
should be something like: 

```
this.destination.set(provider(new Callable<File>() { 
    File call() { 
        return resolveToFile(destination);
    }
}))
```


or, even better:

```
this.destination.set(project.toFileProvider(destination));
```

where we have: `Provider<File> project.toFileProvider(Object f)`